src/main/webapp/app/participation-websocket/participation-websocket.module.ts

```diff
@@ -0,0 +1,16 @@
+import { NgModule } from '@angular/core';
+
+import { ArtemisSharedModule } from 'app/shared';
+import { ParticipationWebsocketService } from 'app/participation-websocket/participation-websocket.service';
+
+@NgModule({
+    imports: [ArtemisSharedModule],
+})
+export class ArtemisParticipationWebsocketModule {
+    static forRoot() {
```

Comment:
This could also be done in the same module as before.

Reply:
Could you provide a working solution for this approach?
For this does not work the same way (will still spawn two services):

```
@NgModule({
    imports: [ArtemisSharedModule, RouterModule.forChild(ENTITY_STATES), SortByModule, ArtemisExerciseScoresModule, ArtemisProgrammingExerciseActionsModule],
    declarations: [
        ParticipationComponent,
        ParticipationDeleteDialogComponent,
        ParticipationDeletePopupComponent,
        ParticipationCleanupBuildPlanDialogComponent,
        ParticipationCleanupBuildPlanPopupComponent,
    ],
    entryComponents: [
        ParticipationComponent,
        ParticipationDeleteDialogComponent,
        ParticipationDeletePopupComponent,
        ParticipationCleanupBuildPlanDialogComponent,
        ParticipationCleanupBuildPlanPopupComponent,
    ],
    providers: [ParticipationService, ParticipationPopupService, { provide: JhiLanguageService, useClass: JhiLanguageService }],
})
export class ArtemisParticipationModule {
    static forRoot() {
        return {
            ngModule: ArtemisParticipationModule,
            providers: [ParticipationWebsocketService],
        };
    }
    constructor(private languageService: JhiLanguageService, private languageHelper: JhiLanguageHelper) {
        this.languageHelper.language.subscribe((languageKey: string) => {
            if (languageKey) {
                this.languageService.changeLanguage(languageKey);
            }
        });
    }
}
```

Reply:
I need to look into this tomorrow.

Reply:
Thank you

Reply:
@jpbernius Did you have time to look into this yet? Or should we test & merge this the way it is?

Reply:
@jpbernius Works now, thanks!